File: app/src/main/java/com/cabe/app/novel/parser/BaseParser.java
package com.cabe.app.novel.parser;

import com.cabe.app.novel.model.BaseObject;

import org.jsoup.nodes.Document;

/**
 * 作者：沈建芳 on 2017/10/16 21:08
 */
public interface BaseParser<T extends BaseObject> {
    T parserDoc(Document doc);
}

File: app/src/main/java/com/cabe/app/novel/parser/BaseParser.kt
package com.cabe.app.novel.parser

import com.cabe.app.novel.model.BaseObject
import org.jsoup.nodes.Document

/**
 * 作者：沈建芳 on 2017/10/16 21:08
 */
interface BaseParser<T : BaseObject?> {
    fun parserDoc(doc: Document?): T
}